core/scheduler/scheduler.go

```diff
@@ -277,7 +280,8 @@ func (s *Scheduler) resolveDuties(ctx context.Context, slot core.Slot) error {
 		return err
 	}
 
-	s.setResolvedEpoch(uint64(slot.Epoch()))
+	s.setResolvedEpoch(slot.Epoch())
+	s.trimDuties(slot.Epoch() - trimEpochOffset)
```

Comment:
trim when resolved so only need to check `resolvedEpoch` to know when things were trimmed